### PR TITLE
Feature/mvvm implementation

### PR DESCRIPTION
**Pull request by Srishtikumari9**

### CODE REVIEW

File: app/src/main/java/com/example/socialapp/SharedPreferenceHelper.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+public class SharedPreferenceHelper{
+    static void setSharedPreferenceString(Context context, String key, String value){
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
+        SharedPreferences.Editor editor = preferences.edit();
+        editor.putString(key, value);
+        editor.apply();
+    }
+    static String getSharedPreferenceString(Context context, String key, String defValue){
```
Comment (pawankgupta-se) [inline app/src/main/java/com/example/socialapp/SharedPreferenceHelper.java]: We we can have shorter name like `getString`.

File: app/src/main/java/com/example/socialapp/SharedPreferenceHelper.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.example.socialapp;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.preference.PreferenceManager;
+
+
+public class SharedPreferenceHelper{
+    static void setSharedPreferenceString(Context context, String key, String value){
```
Comment (pawankgupta-se) [inline app/src/main/java/com/example/socialapp/SharedPreferenceHelper.java]: we can have shorter name like `setString`

Comment (pawankgupta-se) [reply app/src/main/java/com/example/socialapp/SharedPreferenceHelper.java]: Method should be public.

File: app/src/main/java/com/example/socialapp/SharedPreferenceHelper.java (hunk 1 above)
Comment (pawankgupta-se) [inline app/src/main/java/com/example/socialapp/SharedPreferenceHelper.java]: Method should be public.

File: app/src/main/java/com/example/socialapp/NextActivity.java [hunk 3]
```diff
@@ -42,20 +50,19 @@ public void onClick(View view) {
...
             public void onResponse(Call<AccessTokenResponse> call, Response<AccessTokenResponse> response) {
                 runOnUiThread(new Runnable() {
                     @Override
                     public void run() {
                         String accessToken = response.body().getAccessToken();
-                        SharedPreferences.Editor editor = getSharedPreferences("LetsConnect", MODE_PRIVATE).edit();
-                        editor.putString("accessToken", accessToken);
-                        editor.apply();
+                        SharedPreferenceHelper.setSharedPreferenceString(NextActivity.this, PREF_KEY_ACCESS_TOKEN, accessToken);
+                        SharedPreferenceHelper.getSharedPreferenceString(NextActivity.this, PREF_KEY_ACCESS_TOKEN, "");
```
Comment (pawankgupta-se) [inline app/src/main/java/com/example/socialapp/NextActivity.java]: Why we need this line, if we don't want to fethch token here?

File: app/src/main/java/com/example/socialapp/ui/posts/PostsActivity.java [hunk 4]
```diff
@@ -23,38 +26,41 @@
...
         superListView.setAdapter(arr);
 
         getPosts();
     }
+
     private void getPosts() {
-        PostsRepository postsRepository = new PostsRepositoryImpl();
-        SharedPreferences prefs = getSharedPreferences("LetsConnect", MODE_PRIVATE);
-        String accessToken = prefs.getString("accessToken", "");
+        SharedPreferences prefs = getSharedPreferences(SHARED_PREFERENCE_NAME, MODE_PRIVATE);
```
Comment (pawankgupta-se) [inline app/src/main/java/com/example/socialapp/ui/posts/PostsActivity.java]: This line is not needed any more.

**Review by pawankgupta-se — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/socialapp/SharedPreferenceHelper.java (hunk 1 above)
Comment (Srishtikumari9) [reply app/src/main/java/com/example/socialapp/SharedPreferenceHelper.java]: Implemented

File: app/src/main/java/com/example/socialapp/SharedPreferenceHelper.java (hunk 2 above)
Comment (Srishtikumari9) [reply app/src/main/java/com/example/socialapp/SharedPreferenceHelper.java]: Implemented

File: app/src/main/java/com/example/socialapp/NextActivity.java (hunk 3 above)
Comment (Srishtikumari9) [reply app/src/main/java/com/example/socialapp/NextActivity.java]: Implemented

File: app/src/main/java/com/example/socialapp/ui/posts/PostsActivity.java (hunk 4 above)
Comment (Srishtikumari9) [reply app/src/main/java/com/example/socialapp/ui/posts/PostsActivity.java]: Implemented

**Review by pawankgupta-se — APPROVED** [review]

**Pull request merged**